lib/3scale/api/client.rb

```diff
@@ -338,6 +347,38 @@ def create_pricingrule(application_plan_id, metric_id, attributes)
         extract(entity: 'pricing_rule', from: response)
       end
 
+      # @api public
+      # @return [Array<Hash>]
+      def list_activedocs
+        response = http_client.get('/admin/api/active_docs')
+        extract(collection: 'api_docs', entity: 'api_doc', from: response)
+      end
+
+      # @api public
+      # @param [Hash] attributes ActiveDocs attributes
+      # @return [Hash]
+      def create_activedocs(attributes)
+        response = http_client.post('/admin/api/active_docs', body: attributes)
+        extract(entity: 'api_doc', from: response)
+      end
+
+      # @api public
+      # @param [Fixnum] id ActiveDocs ID
+      # @param [Hash] attributes ActiveDocs attributes
+      # @return [Hash]
+      def update_activedocs(id, attributes)
+        response = http_client.put("/admin/api/active_docs/#{id}", body: attributes)
+        extract(entity: 'api_doc', from: response)
+      end
+
+      # @api public
+      # @return [Hash]
+      def delete_activedocs(id)
+        raise 'empty id' if id.nil?
```

Comment:
ids are `Integer`

```json
{
    "id": 95510,
    "system_name": "myAcSN7",
    "name": "myActiveDocs3",
    "description": "some description",
    "published": false,
    "skip_swagger_validations": true,
    "body": "---\nswagger: \"2.0\"\ninfo:\n  title: \"mytitle\"\n  description: \"Mydescr\"\n  version: \"1.0.0\"\nbasePath: \"/v2\"\npaths:\n  /pet:\n    post:\n      operationId: \"addPet\"\n      description: \"\"\n",
    "service_id": 2555417769646,
    "created_at": "2019-03-03T18:42:10+01:00",
    "updated_at": "2019-03-03T18:42:10+01:00"
  }
```

Reply:
I think even being an integer we would have the same problem. The method could receive a string without problem, and what is sent to the API is a string really.

However let's wait a moment... I did not want you to perform this changes without questioning :) . I've seen that all other delete methods do not have parameter checking so maybe would be better to keep following that consistency and do not at parameter checking for this changes either.

I think there's not a "good answer". I see it as a tradeoff:
 * If we add parameter checking (duplicate parameter checking, being it in the client and the API itself) and the API that the methods use change then we might need to update the methods -> more difficult to maintain
 * If we do not add parameter checking we delegate the responsability of a good input to the destination API, risking the possibility of causing some unintended action due to a bad input if when destination API does not check some edge cases correctly.

I'd like to discuss this with more people because it's a recurring doubt that I have with some of our API clients, but I don't want to block this PR.
I'd say that continuing the current "consistency" with how the other methods treated parameter checking (this is, not doing parameter checking) would be good now as with your original PR content for the new methods you added.

Reply:
Be conservative in what you send, be liberal in what you accept

Reply:
I can add `id.to_i'` to make sure it will be integer and fail if empty is received.

That would work for you?